runtime/syntax/man.vim

```diff
@@ -5,18 +5,30 @@ if exists('b:current_syntax')
   finish
 endif
 
-syntax case  ignore
-syntax match manReference      display '[^()[:space:]]\+([0-9nx][a-z]*)'
-syntax match manSectionHeading display '^\S.*$'
-syntax match manTitle          display '^\%1l.*$'
-syntax match manSubHeading     display '^ \{3\}\S.*$'
-syntax match manOptionDesc     display '^\s\+\%(+\|-\)\S\+'
+syntax match manBackspacedCharacter display conceal '.\b'
+syntax match manBold                display contains=manBackspacedCharacter '\%(\([[:graph:]]\)\b\1\)\+'
+syntax match manUnderline           display contains=manBackspacedCharacter '\%(_\b[^_]\)\+'
 
-highlight default link manTitle          Title
-highlight default link manSectionHeading Statement
-highlight default link manOptionDesc     Constant
-highlight default link manReference      PreProc
-highlight default link manSubHeading     Function
+if !exists('#man_highlight_groups')
+  function! s:init_highlight_groups() abort
+    let group = 'Keyword'
+    while 1
+      let values = execute('highlight '.group)
+      if values =~# '='
+        let values = substitute(values, '.* \(\w\+=.*\)', '\1', '')
+        break
+      endif
+      let group = substitute(values, '.* to \(\w\+\)', '\1', '')
```

Comment:
Instead of this hacky stuff, should I just make manBold link to Keyword and not bold it by default? If a user wants to bold it, they just have to define manBold themselves. Would be much easier.